spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/ConfigMapPropertySource.java

```diff
@@ -89,12 +97,34 @@ private static String getName(KubernetesClient client, String name,
 	}
 
 	private static Map<String, Object> getData(KubernetesClient client, String name,
-			String namespace, Environment environment) {
+			String namespace, Environment environment, boolean enableVersioning,
+			Map<String, String> labels) {
 		try {
 			Map<String, Object> result = new LinkedHashMap<>();
-			ConfigMap map = StringUtils.isEmpty(namespace)
```

Comment:
This is heading in the right direction but not exactly what I was envisioning.

We should allow the app to provide filters for both configmaps and secrets.  These filters can be just simple lambdas which take in a configmap or secret and return a boolean if it matches or doesn't.  This would eliminate all the versioning code here.  Maybe we could provide a versioning filter if we wanted to that apps could reuse, but it would not be part of the ConfigMapPropertySource code.  Does that make sense?

Reply:
Yes, definitely it has sense. I'll think about it.

Reply:
I removed the whole code for versioning, and just leave that for labelling support. If we want provide a filter we should probably use AOP (?), which is currently not used in that project.